addons/fleet/models/fleet_vehicle.py

```diff
@@ -200,36 +198,24 @@ def _compute_contract_reminder(self):
         for record in self:
             overdue = False
             due_soon = False
-            total = 0
-            name = ''
             state = ''
-            for element in record.log_contracts:
-                if element.state in ('open', 'expired') and element.expiration_date:
-                    current_date_str = fields.Date.context_today(record)
-                    due_time_str = element.expiration_date
-                    current_date = fields.Date.from_string(current_date_str)
-                    due_time = fields.Date.from_string(due_time_str)
-                    diff_time = (due_time - current_date).days
-                    if diff_time < 0:
-                        overdue = True
-                        total += 1
-                    if diff_time < delay_alert_contract:
-                        due_soon = True
-                        total += 1
-                    if overdue or due_soon:
-                        log_contract = self.env['fleet.vehicle.log.contract'].search([
-                            ('vehicle_id', '=', record.id),
-                            ('state', 'in', ('open', 'expired'))
-                            ], limit=1, order='expiration_date asc')
-                        if log_contract:
-                            # we display only the name of the oldest overdue/due soon contract
-                            name = log_contract.name
-                            state = log_contract.state
+            log_contract = self.env['fleet.vehicle.log.contract'].search(
+                [
+                    ('vehicle_id', '=', record.id),
+                    ('state', 'in', ('open', 'expired')),
+                ],
+                limit=1, order='expiration_date desc')
+
+            if log_contract.expiration_date:
```

Comment:
This works now, as it above you're only querying the open and expired contracts, but if you also look for new contracts, odds are the contract with the expiration date furthest in the future will be new, while if the current contract is still running, I would argue that you'd want to display the running state, instead of the new one

Reply:
Yeah I forgot to take into account the future ones thanks a lot ^^ but like gmf said we need to just take it like an ATM so only the last contract (by exp date) should be taken into account for the warning because it's set by human

But I would argue that it would be a nice addition to add like a fa-copy Icon (just because it represent two pieces of paper) in blue or yellow if there is a gap between the last contract exp date and the next future contract start date. So like that we could make the fleet manager aware of potentially sub-optimal usage of the car (or a human error) and maybe change the color of it to orange if the driver is the same in the 2 contracts because for me it's a bit strange to have the same driver for the same car with no contract for like a month

Reply:
Yes for the warnings I agree, just take the one furthest in the future.B ut the field contract_state indicates what the state of the current (or the next one if there currently is no contract, or the previous if there is no current nor next one) contract is. For this field to be 'new' when there is currently a contract running is a bit weird to me.

And for your last point indeed it's a bit strange, but Geoffrey's stance here is that as long as the fleet manager is made aware that contracts are about to expire it's up to them how to solve it and if they made a contract two months in the future they probably have a reason for that. If we'd handle this differently I think it would be more useful to, instead of warning icon, make the warning pop up when you make the new contract, or to even add a constraint. But for the moment Geoffrey doesn't think it's worth it.